pywatts/core/step.py

```diff
@@ -75,13 +81,19 @@ def _fit(self, inputs: Dict[str, BaseStep], target_step):
     def _callbacks(self):
         # plots and writs the data if the step is finished.
         for callback in self.callbacks:
+            dim = _get_time_indexes(self.buffer)[0]
+
+            to_plot = self.buffer
+            if self.current_run_setting.online_start is not None:
+                to_plot = {k: self.buffer[k][self.buffer[k][dim] >= self.current_run_setting.online_start] for k in
+                           self.buffer.keys()}
             if isinstance(callback, BaseCallback):
                 callback.set_filemanager(self.file_manager)
             if isinstance(self.buffer, xr.DataArray) or isinstance(self.buffer, xr.Dataset):
                 # DEPRECATED: direct DataArray or Dataset passing is depricated
                 callback({"deprecated": self.buffer})
```

Comment:
`self.buffer` -> `to_plot` ?

Reply:
If not online_start, than the complete buffer should be plotted. If it has an online start only the data after the online_start should be plotted. I reformatted it a little bit to make it clearer.